src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -4326,62 +4326,39 @@ public static final void setArenaWins(final int wins) {
    * @return familiar The first familiar matching this race
    */
   public static final FamiliarData findFamiliar(final String race) {
-    if (FamiliarData.NO_FAMILIAR.getRace().equals(race)) {
-      return FamiliarData.NO_FAMILIAR;
-    }
-
-    // Don't even look if you are an Avatar
-    if (KoLCharacter.inAxecore() || KoLCharacter.isJarlsberg()) {
-      return null;
-    }
-
-    FamiliarData[] familiarArray = new FamiliarData[KoLCharacter.familiars.size()];
-    KoLCharacter.familiars.toArray(familiarArray);
-
-    for (int i = 0; i < familiarArray.length; ++i) {
-      FamiliarData familiar = familiarArray[i];
-      if (familiar.getRace().equals(race)) {
-        return familiar;
-      }
-    }
-
-    return null;
+    return findFamiliar(f -> f.getRace().equalsIgnoreCase(race));
   }
 
+  /**
+   * Accessor method to find the specified familiar.
+   *
+   * @param familiarId The id of the familiar to find
+   * @return familiar The first familiar matching this id
+   */
   public static final FamiliarData findFamiliar(final int familiarId) {
-    if (familiarId == -1) {
-      return FamiliarData.NO_FAMILIAR;
-    }
+    return findFamiliar(f -> f.getId() == familiarId);
+  }
+
+  private static final FamiliarData findFamiliar(final Predicate<FamiliarData> familiarFilter) {
+    // Quick check against NO_FAMILIAR
+    if (familiarFilter.test(FamiliarData.NO_FAMILIAR)) return FamiliarData.NO_FAMILIAR;
 
     // Don't even look if you are an Avatar
-    if (KoLCharacter.inAxecore() || KoLCharacter.isJarlsberg() || KoLCharacter.isSneakyPete()) {
-      return null;
-    }
+    if (KoLCharacter.getPath().isAvatar) return null;
```

Comment:
While a good idea, this doesn't mean what you think it means ;)

To be more precise, some avatar paths, like Plumber, have familiars. I'd recommend splitting out another boolean in the `Path` constructor.

Reply:
Ah yeah good point